class_15_sockets/homework/client.py

```diff
@@ -0,0 +1,64 @@
+import socket
+import time
+
+import psutil
+import pyautogui
+
+from metric import Metric
+
+HOST = "localhost"
+PORT = 9999
+TIME_FORMAT = "%m/%d/%Y %H:%M:%S"
+SLEEP_TIME = 10
+
+
+class ConnException(Exception):
+    pass
+
+
+def client_send(host, port, data):
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect((host, port))
+        print(f"Connected to {host}:{port}")
+    except ConnectionError:
+        print("Connection error occurred")
+        return
+
+    time_ = time.strftime(TIME_FORMAT, time.localtime())
+    data = f"{time_}: {data}".encode("utf-8")
+    with sock:
+        msg_len = len(data)
+        print("sending data:", data)
+        while msg_len:
```

Comment:
A good example of sending data via little chunks, another example: https://docs.python.org/3/howto/sockets.html#using-a-socket

Reply:
Oh, I think I see the issue. Instead of trying to send remaining part of bytes I still keep sending whole bunch of data.
making a fix here.